bbc/simorgh
# WSTEAM1-399 - Related content display with no images

Resolves JIRA WSTEAM1-399

Overall changes
======
- Allows the related content section for Article and MediaArticle pages to display text-only related content items
- This is an issue that is present on PS services that we serve AMP for
- Elevates the `RelatedContentSection` component to the `components` directory and removes the individual usages from Article and MediaArticle pages in favour of this single component

Code changes
======
- Refactors the Related Content section component to a single location that Article and MediaArticle pages can use
- Adds support for text-only related content items by checking if the headline appears first in the Optimo blocks, extracting the content and then applying styling as appropriate
- Moves `EventTrackingData` into a new `eventTracking.ts` type file and updates references to this type throughout

Testing
======
1. Visit http://localhost:7080/news/articles/c25ppwql5qvo?renderer_env=live and confirm that the single related content item appears correctly
2. Visit http://localhost:7080/naidheachdan/sgeulachdan/c18q7nedn2ko.amp and confirm that multiple related content items appear correctly
3. Visit http://localhost:7080/portuguese/articles/cqezz876rq7o?renderer_env=live and confirm that related content items with image continue to appear correctly

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
